src/OpenTracing/Propagation/TextMapInjectAdapter.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace OpenTracing.Propagation
+{
+    /// <summary>
+    /// A TextMap carrier for use with <see cref="ITracer.Inject{TCarrier}"/> ONLY (it has no read methods). Note that
+    /// the TextMap interface can be made to wrap around arbitrary data types (not just Dictionary{string, string} as
+    /// illustrated here).
+    /// </summary>
+    /// <seealso cref="ITracer.Inject{TCarrier}"/>
+    public sealed class TextMapInjectAdapter : TextMap
+    {
+        private readonly IDictionary<string, string> _dictionary;
+
+        public TextMapInjectAdapter(IDictionary<string, string> dictionary)
+        {
+            _dictionary = dictionary;
+        }
+
+        public IEnumerator<KeyValuePair<string, string>> GetEnumerator()
+        {
+            throw new InvalidOperationException(
+                $"{nameof(TextMapInjectAdapter)} should only be used with {nameof(ITracer)}.{nameof(ITracer.Inject)}");
+        }
+
+        public void Set(string key, string value)
+        {
+            _dictionary.Add(key, value);
```

Comment:
This should be `_dictionary[key] = value;` to ensure existing values are overwritten if it is called multiple times. Java's `put` has the same behavior.